packages/flutter_tools/lib/src/asset.dart

```diff
@@ -317,7 +319,9 @@ DevFSContent _createFontManifest(Map<String, dynamic> manifestDescriptor,
     fonts.addAll(manifestDescriptor['fonts']);
   if (fonts.isEmpty)
     return null;
-  return new DevFSStringContent(JSON.encode(fonts));
+  String json = JSON.encode(fonts);
+  json = json.replaceAll(_assetJsonLeadingParentRegExp, '"asset":"');
```

Comment:
Ya, I struggled on this too and puked some blood as I clicked git push...
Though the problem is manifestDescriptor (a bit above) is a YamlMap that's unmodifiable and might have some particular bevahiours. Either way some conversion needed to be done before modifying it.